File: src/main/java/com/viaversion/aas/codec/packet/Packet.java
package com.viaversion.aas.codec.packet;

import io.netty.buffer.ByteBuf;
import org.jetbrains.annotations.NotNull;

/**
 * A mutable object which represents a Minecraft packet data
 */
public interface Packet {
	void decode(@NotNull ByteBuf byteBuf, int protocolVersion) throws Exception;

	void encode(@NotNull ByteBuf byteBuf, int protocolVersion) throws Exception;
}

File: src/main/kotlin/com/viaversion/aas/codec/packet/Packet.kt
package com.viaversion.aas.codec.packet

import io.netty.buffer.ByteBuf

/**
 * A mutable object which represents a Minecraft packet data
 */
interface Packet {
    fun decode(byteBuf: ByteBuf, protocolVersion: Int)
    fun encode(byteBuf: ByteBuf, protocolVersion: Int)
}